sklearn/utils/estimator_checks.py

```diff
@@ -395,57 +402,143 @@ def _yield_all_checks(estimator, legacy: bool):
         yield check_fit_non_negative
 
 
-def _maybe_mark_xfail(estimator, check, pytest):
-    # Mark (estimator, check) pairs as XFAIL if needed (see conditions in
-    # _should_be_skipped_or_marked())
-    # This is similar to _maybe_skip(), but this one is used by
-    # @parametrize_with_checks() instead of check_estimator()
+def _check_name(check):
+    if hasattr(check, "__wrapped__"):
+        return _check_name(check.__wrapped__)
+    return check.func.__name__ if isinstance(check, partial) else check.__name__
+
 
-    should_be_marked, reason = _should_be_skipped_or_marked(estimator, check)
-    if not should_be_marked:
+def _maybe_mark(
+    estimator,
+    check,
+    expected_failed_checks: dict[str, str] | None = None,
+    mark: Literal["xfail", "skip", None] = None,
+    pytest=None,
+):
+    """Mark the test as xfail or skip if needed.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        Estimator instance for which to generate checks.
+    check : partial or callable
+        Check to be marked.
+    expected_failed_checks : dict[str, str], default=None
+        Dictionary of the form {check_name: reason} for checks that are expected to
+        fail.
+    mark : "xfail" or "skip" or None
+        Whether to mark the check as xfail or skip.
+    pytest : pytest object, default=None
+        Pytest module to use to mark the check.
```

Comment:
Sorry for my misunderstanding. I don't understand how this is actually used. Do you mind explaining why we would have None sometimes instead of pytest module itself?

Reply:
`check_estimator` doesn't need `pytest`, while `parametrize_with_checks` does. I expanded the comment in the docstring to explain.